internal/pkg/agent/application/upgrade/step_unpack_test.go

```diff
@@ -190,7 +190,7 @@ func TestUpgrader_unpackTarGz(t *testing.T) {
 			archiveFile, err := tt.args.archiveGenerator(t, tt.args.archiveFiles)
 			require.NoError(t, err, "creation of test archive file failed")
 
-			got, err := untar(log, archiveFile, testDataDir)
+			got, err := untar(log, archiveFile, testDataDir, false, "")
```

Comment:
You should be able to unit test that the logic to respect the flavors when unpacking actually works.